Redo the overhead analysis (#95)
sframe-wg/sframe: draft-ietf-sframe-enc.md

contributors to the document. His passion and energy were key to the design and
development of SFrame.

> # Overhead Analysis

Reviewer: This is certainly a better title, but breaks building due to [line 201](https://github.com/sframe-wg/sframe/blob/c0c80e083563ed5cedbe02b7d357d9a2c5b4e70f/draft-ietf-sframe-enc.md?plain=1#L201) `{{overhead}}` not being changed to `{{overhead-analysis}}`

Any use of SFrame will impose overhead in terms of the amount of bandwidth

> necessary to transmit a given media stream.  Exactly much overhead will be added

Reviewer: ```suggestion
necessary to transmit a given media stream.  Exactly how much overhead will be added
```

> depend on several factors:

Reviewer: ```suggestion
depends on several factors:
```

* How many senders are involved in a conference (length of KID)
* How long the conference has been going on (length of CTR)
* The ciphersuite in use (length of authentication tag)
* Whether SFrame is used to encrypt packets, whole frames, or some other unit

Overall, the overhead rate in kilobits per second can be estimated as:

```
OverheadKbps = (1 + |CTR| + |KID| + |TAG|) * 8 * UnitsPerSecond / 1024
```

> Here the constant value `1` reflect the fixed SFrame header; `|CTR|` and

Reviewer: ```suggestion
Here the constant value `1` reflects the fixed SFrame header size in bytes; `|CTR|` and
```

Reviewer: The "size in bytes" suggestion could probably be omitted.

> `|KID|` reflect the length of those fields; `|TAG|` reflects the cipher

Reviewer: ```suggestion
`|KID|` reflect the lengths of those fields; `|TAG|` reflects the cipher
```

Reviewer: Actually, this one might be subjective...

overhead; and `UnitsPerSecond` reflects the number of SFrame-encrypted units
sent per second (e.g., packets or frames per second).

In the remainder of this secton, we compute overhead estimates for a collection
of common scenarios.

## Assumptions

In the below calculations, we make conservative assumptions about SFrame
overhead, so that the overhead amounts we compute here are likely to be an upper
bound on those seen in practice.

| Field           | Bytes | Explanataion                                      |
|:----------------|------:|:--------------------------------------------------|
| Fixed header    | 1     | Fixed                                             |
| Key ID (KID)    | 2     | >255 senders; or MLS epoch (E=4) and >16 senders  |
| Counter (CTR)   | 3     | More than 24 hours of media in common cases       |
| Cipher overhead | 16    | Full GCM tag (longest defined here)               |

In total, then, we assume that each SFrame encryption will add 22 bytes of
overhead.

> We consider two "units" of encryption, frames and packets.  In each scenario, we

Reviewer: We call them "units", then refer to them as scenarios immediatel afterward. Should "units" be scenarios?
```suggestion
We consider two scenarios of encryption, frames and packets.  In each scenario, we
```

Reviewer: Also wondering on this, is the comma correct? I suppose without using the oxford comma it does indeed describe the 2 scenarios, but I wonder if this is instead a good time to use a colon:
```suggestion
We consider two "units" of encryption: frames and packets.  In each scenario, we
```

compute the SFrame overhead in absolute terms (Kbps) and as a percentage of the
base bandwidth.

## Audio

In audio streams, there is typically a one-to-one relationship between frames
and packets, so the overhead is the same whether one uses SFrame at a per-packet

> or whole-frame level.

Reviewer: We refer to this as "per-frame" later, maybe we should here too, to keep consistency with other usages, as well as the naming convention per-*.
```suggestion
or per-frame level.
```

| 640 x 360   |  30 |       400 |           5.2 |       1.3% |
| 1280 x 720  |  30 |      1500 |           5.2 |       0.3% |
| 1920 x 1080 |  60 |      7200 |          10.3 |       0.1% |
{: #video-overhead-per-frame title="SFrame overhead for a video stream encrypted per-frame" }

| Scenario    | fps | pps | Base Kbps | Overhead Kbps | Overhead % |
|:------------|:---:|:---:|:---------:|:-------------:|:----------:|
| 426 x 240   | 7.5 | 7.5 |        45 |           1.3 |       2.9% |
| 640 x 360   |  15 |  30 |       200 |           5.2 |       2.6% |
| 640 x 360   |  30 |  60 |       400 |          10.3 |       2.6% |
| 1280 x 720  |  30 | 180 |      1500 |          30.9 |       2.1% |
| 1920 x 1080 |  60 | 780 |      7200 |         134.1 |       1.9% |
{: #video-overhead-per-packet title="SFrame overhead for a video stream encrypted per-packet" }

In the per-frame case, the SFrame percentage overhead approaches zero as the
quality of the video goes up, since bandwidth is driven more by picture size
than frame rate.  In the per-packet case, the SFrame percentage overhead
approaches the ratio between the SFrame overhead per packet and the MTU (here 22

> bytes of SFrame overhead divided by an assumed 1200-byte MTU, or about 1.8%).

Reviewer: ```suggestion
here a 22-byte SFrame overhead divided by an assumed 1200-byte MTU, or about 1.8%).
```
```suggestion
bytes of SFrame overhead divided by an assumed 1200 byte of MTU, or about 1.8%).
```
Again, more of a style choice, this makes sense as is.

## Conferences

Real conferences usually involve several audio and video streams.  The overhead
of SFrame in such a conference is the aggregate of the overhead over all the

> individual streams.  Thus, while SFrame incur a large percentage overhead on an

Reviewer: ```suggestion
individual streams.  Thus, while SFrames incur a large percentage overhead on an
```
or
```suggestion
individual streams.  Thus, while SFrame incurs a large percentage overhead on an
```

audio stream, if the conference also involves a video stream, then the audio
overhead is likely negligible relative to the overall bandwidth of the
conference.

For example, {{conference-overhead}} shows the overhead estimates for a two

> person conference where one person is sending low-quality media and the other

Reviewer: ```suggestion
person conference, where one person is sending low-quality media and the other
```
